### PR TITLE
2375 convergence plots

### LINKED ISSUE #2375: Add Convergence Plot to AsyncTaskDialog Progress Menu Bar
### Desired Behaviour

Migrate and cleanup implementation of convergence plot developed as part of #2369 

Cleanup should creating a template widget in the UI file to contain the convergence plot so that when a convergence plot is added to progress menu, It is placed above the cancel button implemented as part of #2370 

### Current Behaviour

Convergence plots are not used within the AsyncTaskDialog progress menu bar within Mantid Imaging.

### PR DESCRIPTION
**Pull request by JackEAllen**
### Issue

Closes #2375 

### Description

*Add a description of the changes made*.
Expose Reconstruction Iterations and Losses through MIProgressCallback to AsyncTaskDialog progress bar to display convergence plot.

### Testing 

*Describe the tests that were used to verify your changes*.
* Manually tested
* Updated unit tests

### Acceptance Criteria 

*How should the reviewer test your changes*?
* Unit tests pass
* Convergence plot is displayed when reconstructing using PDHG.

### Documentation

*How have you changed the documentation to reflect your changes? All changes should be noted in the appropriate file in docs/release_notes*

`docs/release_notes/next/feature-2375-display-convergence-plot-during-reconstruction`

### CODE REVIEW

File: mantidimaging/gui/dialogs/async_task/view.py [hunk 1]
```diff
@@ -63,6 +69,10 @@ def set_progress(self, progress: float, message: str):
         # Update progress bar
         self.progressBar.setValue(int(progress * 1000))
 
+    def set_progress_plot(self, x, y):
```
Comment (samtygier-stfc) [inline mantidimaging/gui/dialogs/async_task/view.py:72]: Type annotations

File: mantidimaging/gui/dialogs/async_task/presenter.py [hunk 2]
```diff
@@ -62,10 +64,19 @@ def do_start_processing(self) -> None:
     def task_is_running(self) -> None:
         return self.model.task_is_running
 
+    def update_progress_plot(self, iterations, losses) -> None:
```
Comment (samtygier-stfc) [inline mantidimaging/gui/dialogs/async_task/presenter.py:67]: Type annotations

**Review by samtygier-stfc — CHANGES_REQUESTED** [review]
Looks good and works locally.

There are some new methods that could have type annotations.

There are also some possible optimisations, but I'd be happy for them to be in a future PR.

- Only create the `PlotWidget` the first time `set_progress_plot()` is called
- Only call `update_progress_plot()` when there is something new to plot (the loss is only calculated every 10th iteration). (maybe if the last element in `iterations` is the same as the current step, or something like that).

File: mantidimaging/gui/dialogs/async_task/view.py (hunk 1 above)
Comment (JackEAllen) [reply mantidimaging/gui/dialogs/async_task/view.py:72]: Thanks. Had meant to add these and completely forgot

**Review by samtygier-stfc — APPROVED** [review]

**Pull request merged**